game-backend/src/game-engine/mechanics/powerup-spawner/powerups/healthRegenBoost.test.ts

```diff
@@ -0,0 +1,74 @@
+import { State, GameObjectTypes, UserContext } from '@darwin/types';
+import StateBuilder from '../../../../test-helper/StateBuilder';
+import { getUnit } from '../../../../helper/gameObjects';
+import { getConsumable } from '../../../../helper/consumable';
+import { ATTRIBUTE_BOUNDARIES } from '../../../../helper/attribute';
+import consumeHealthRegenBoost from './healthRegenBoost';
+
+describe('enduranceBoost powerup consumption', () => {
+  const NORMAL_UNIT_ID = 'unit1';
+  const MAXED_UNIT_ID = 'unit2';
+  const POWERUP_ID = 'powerup1';
+
+  const normalUserContext: UserContext = {
+    unitId: NORMAL_UNIT_ID,
+  };
+  const maxedUserContext: UserContext = {
+    unitId: MAXED_UNIT_ID,
+  };
+
+  const state: State = StateBuilder.buildState()
+    .addUnit({
+      id: NORMAL_UNIT_ID,
+      x: 1,
+      y: 1,
+      attributes: { healthRegenBoost: 0 },
+    })
+    .addUnit({
+      id: MAXED_UNIT_ID,
+      x: 2,
+      y: 2,
+      attributes: {
+        healthRegenBoost: ATTRIBUTE_BOUNDARIES.healthRegenBoost.max,
+      },
+    })
+    .addPowerup({
+      id: POWERUP_ID,
+      x: 3,
+      y: 3,
+      type: GameObjectTypes.HealthRegenBoost,
+    })
+    .build();
+
+  it('should enhance units attribute when consumed and remove powerup', () => {
+    const newState = consumeHealthRegenBoost(
+      POWERUP_ID,
+      state,
+      normalUserContext
+    );
+
+    const unit = getUnit(newState, NORMAL_UNIT_ID);
+    const powerup = getConsumable(newState, POWERUP_ID);
+
+    expect(unit.attributes.healthRegenBoost).toBe(10);
+    expect(powerup).toBeUndefined();
+    expect(newState.objectIds).not.toContain(POWERUP_ID);
+  });
+
+  it('should respect attribute boundaries and not remove powerup', () => {
```

Comment:
in the test descriptions, one time is written 'remove powerup' and the second time is written 'not remove powerup'., yet in both test cases you check for: expect(powerup).toBeUndefined(); ?

Reply:
i see `toBeDefined` on line 71

Reply:
That's right but the test is `expect(powerup).toBeDefined();` and `expect(powerup).toBe`**Un**`defined();` so it should be fine?

Reply:
yes i was wrong (facepalm emoji)